src/lib/statistics/join_graph_statistics_cache.cpp

```diff
@@ -19,13 +19,13 @@
 namespace hyrise {
 
 JoinGraphStatisticsCache JoinGraphStatisticsCache::from_join_graph(const JoinGraph& join_graph) {
-  VertexIndexMap vertex_indices;
-  for (auto vertex_idx = size_t{0}; vertex_idx < join_graph.vertices.size(); ++vertex_idx) {
+  const auto vertex_count = join_graph.vertices.size();
+  auto vertex_indices = VertexIndexMap{vertex_count};
+  for (auto vertex_idx = size_t{0}; vertex_idx < vertex_count; ++vertex_idx) {
     vertex_indices.emplace(join_graph.vertices[vertex_idx], vertex_idx);
   }
 
-  PredicateIndexMap predicate_indices;
-
+  auto predicate_indices = PredicateIndexMap{};
```

Comment:
Can' we reserve here as well?

Reply:
I thought the same, but I was not sure about the value to use. The index map is an `ExpressionUnorderedMap` that only depends on the predicates. `join_graph.edges.size()`, however, could be a rough estimation, so let's use that.